From b70c4541741289535042fbeff7f2e3675ec775fc Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Wed, 10 Jun 2020 00:49:14 -0500
Subject: [PATCH 85/92] usb: typec: anx7688: Fix some bugs

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 drivers/usb/typec/anx7688.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/anx7688.c b/drivers/usb/typec/anx7688.c
index 0da14c29c3264..03c589a0a811c 100644
--- a/drivers/usb/typec/anx7688.c
+++ b/drivers/usb/typec/anx7688.c
@@ -377,6 +377,7 @@ static int anx7688_connect(struct anx7688 *anx7688)
                 dev_err(anx7688->dev, "failed to enable vconn\n");
                 goto err_poweroff;
         }
+	anx7688->vconn_on = true;
 
         /* wait till the firmware is loaded (typically ~30ms) */
         for (i = 0; i < 100; i++) {
@@ -496,6 +497,7 @@ static int anx7688_connect(struct anx7688 *anx7688)
 
 err_vconoff:
         regulator_disable(anx7688->supplies[ANX7688_VCONN_INDEX].consumer);
+	anx7688->vconn_on = false;
 err_poweroff:
         anx7688_power_disable(anx7688);
         return ret;
@@ -684,7 +686,7 @@ static int anx7688_update_status(struct anx7688 *anx7688)
 	}
 
 	if (anx7688->vconn_on != vconn_on) {
-		if (vbus_on) {
+		if (vconn_on) {
 			ret = regulator_enable(anx7688->supplies[ANX7688_VCONN_INDEX].consumer);
 			if (ret) {
 				dev_err(anx7688->dev, "failed to enable vconn\n");
-- 
2.26.2

