From a50d2db749614e42aa2751779af844ef87b2d6a8 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Wed, 10 Jun 2020 00:54:15 -0500
Subject: [PATCH 86/92] usb: typec: anx7688: Set role to device on disconnect

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 drivers/usb/typec/anx7688.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/typec/anx7688.c b/drivers/usb/typec/anx7688.c
index 03c589a0a811c..200e33a0b6b5d 100644
--- a/drivers/usb/typec/anx7688.c
+++ b/drivers/usb/typec/anx7688.c
@@ -528,6 +528,8 @@ static void anx7688_disconnect(struct anx7688 *anx7688)
         typec_set_vconn_role(anx7688->port, TYPEC_SINK);
         typec_set_data_role(anx7688->port, TYPEC_DEVICE);
 
+	usb_role_switch_set_role(anx7688->role_sw, USB_ROLE_DEVICE);
+
 	clear_bit(ANX7688_F_CONNECTED, anx7688->flags);
 }
 
-- 
2.26.2

