From 00d9bde7483a8dd166ec3baba59ef12e8e154828 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 9 Jul 2020 17:07:55 +0800
Subject: [PATCH 103/105] usb: typec: anx7688: adjust register settings

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/usb/typec/anx7688.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/typec/anx7688.c b/drivers/usb/typec/anx7688.c
index 2e433b383582..ac1a73570fe9 100644
--- a/drivers/usb/typec/anx7688.c
+++ b/drivers/usb/typec/anx7688.c
@@ -449,7 +449,7 @@ static int anx7688_connect(struct anx7688 *anx7688)
 
 
         /* min/max power in 500 mW units */
-        ret = anx7688_reg_write(anx7688, ANX7688_REG_MAX_POWER, 5 * 2); /* 5 W */
+        ret = anx7688_reg_write(anx7688, ANX7688_REG_MAX_POWER, 15 * 2); /* 5 W */
 	if (ret)
 		goto err_vconoff;
 
@@ -458,8 +458,8 @@ static int anx7688_connect(struct anx7688 *anx7688)
 		goto err_vconoff;
 
 
-        /* auto_pd, try.src, try.sink, goto safe 5V */
-        ret = anx7688_reg_write(anx7688, ANX7688_REG_FEATURE_CTRL, 0x1e);
+        /* auto_pd, try.sink, goto safe 5V */
+        ret = anx7688_reg_write(anx7688, ANX7688_REG_FEATURE_CTRL, 0x1a);
 	if (ret)
 		goto err_vconoff;
 
-- 
2.27.0

