diff -Naur linux-4.11.old/drivers/gpu/drm/sun4i/sun8i_layer.c linux-4.11/drivers/gpu/drm/sun4i/sun8i_layer.c
--- linux-4.11.old/drivers/gpu/drm/sun4i/sun8i_layer.c	2017-05-05 22:45:42.407089271 +0800
+++ linux-4.11/drivers/gpu/drm/sun4i/sun8i_layer.c	2017-05-05 22:50:54.308211874 +0800
@@ -63,6 +63,7 @@
 
 static const uint32_t sun8i_mixer_layer_formats[] = {
 	DRM_FORMAT_RGB888,
+	DRM_FORMAT_ARGB8888,
 	DRM_FORMAT_XRGB8888,
 };
 
diff -Naur linux-4.11.old/drivers/gpu/drm/sun4i/sun8i_mixer.c linux-4.11/drivers/gpu/drm/sun4i/sun8i_mixer.c
--- linux-4.11.old/drivers/gpu/drm/sun4i/sun8i_mixer.c	2017-05-05 22:45:42.407089271 +0800
+++ linux-4.11/drivers/gpu/drm/sun4i/sun8i_mixer.c	2017-05-05 22:54:07.577445946 +0800
@@ -102,6 +102,10 @@
 					     u32 format, u32 *mode)
 {
 	switch (format) {
+	case DRM_FORMAT_ARGB8888:
+		*mode = SUN8I_MIXER_CHAN_UI_LAYER_ATTR_FBFMT_ARGB8888;
+		break;
+
 	case DRM_FORMAT_XRGB8888:
 		*mode = SUN8I_MIXER_CHAN_UI_LAYER_ATTR_FBFMT_XRGB8888;
 		break;
