diff -Naur linux-4.11-rc7.orig/arch/arm/boot/dts/sun8i-h3-nanopi-neo.dts linux-4.11-rc7/arch/arm/boot/dts/sun8i-h3-nanopi-neo.dts
--- linux-4.11-rc7.orig/arch/arm/boot/dts/sun8i-h3-nanopi-neo.dts	2017-04-24 12:52:28.355818201 +0800
+++ linux-4.11-rc7/arch/arm/boot/dts/sun8i-h3-nanopi-neo.dts	2017-04-24 12:59:50.343346823 +0800
@@ -45,4 +45,14 @@
 / {
 	model = "FriendlyARM NanoPi NEO";
 	compatible = "friendlyarm,nanopi-neo", "allwinner,sun8i-h3";
+
+	vdd_cpux: vdd-cpux-fixed1v1 {
+		compatible = "regulator-fixed";
+		regulator-name = "vdd-cpux";
+		regulator-min-microvolt = <1100000>;
+		regulator-max-microvolt = <1100000>;
+		regulator-boot-on;
+		regulator-always-on;
+		gpio = <&r_pio 0 6 GPIO_ACTIVE_LOW>;
+	};
 };
