From 3bae4182c7041d88542a49009da9837e54b959f2 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Wed, 16 Jun 2021 02:06:40 +0800
Subject: [PATCH 3/8] rockchip: pinebook-pro-rk3399: light power LED when boot

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 .../pinebook-pro-rk3399/pinebook-pro-rk3399.c       | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/board/pine64/pinebook-pro-rk3399/pinebook-pro-rk3399.c b/board/pine64/pinebook-pro-rk3399/pinebook-pro-rk3399.c
index 516292aaa5..a6bd5e0d6b 100644
--- a/board/pine64/pinebook-pro-rk3399/pinebook-pro-rk3399.c
+++ b/board/pine64/pinebook-pro-rk3399/pinebook-pro-rk3399.c
@@ -9,10 +9,12 @@
 #include <syscon.h>
 #include <asm/io.h>
 #include <asm/arch-rockchip/clock.h>
+#include <asm/arch-rockchip/gpio.h>
 #include <asm/arch-rockchip/grf_rk3399.h>
 #include <asm/arch-rockchip/hardware.h>
 #include <asm/arch-rockchip/misc.h>
 #include <power/regulator.h>
+#include <dt-bindings/pinctrl/rockchip.h>
 
 #define GRF_IO_VSEL_BT565_SHIFT 0
 #define PMUGRF_CON0_VSEL_SHIFT 8
@@ -53,6 +55,16 @@ static void setup_iodomain(void)
 	rk_setreg(&pmugrf->soc_con0, 1 << PMUGRF_CON0_VSEL_SHIFT);
 }
 
+static void light_power_led(void)
+{
+#define GPIO0_BASE	0xff720000
+	struct rockchip_gpio_regs * const gpio = (void *)GPIO0_BASE;
+
+	int mask = (1UL << RK_PB3);
+	setbits_le32(&gpio->swport_dr, mask);
+	setbits_le32(&gpio->swport_ddr, mask);
+}
+
 int misc_init_r(void)
 {
 	const u32 cpuid_offset = 0x7;
@@ -61,6 +73,7 @@ int misc_init_r(void)
 	int ret;
 
 	setup_iodomain();
+	light_power_led();
 
 	ret = rockchip_cpuid_from_efuse(cpuid_offset, cpuid_length, cpuid);
 	if (ret)
-- 
2.30.2

