From bac9170fb269eba370e5014d3990c07a083dc508 Mon Sep 17 00:00:00 2001
From: Tomas Melin <tomas.melin@vaisala.com>
Date: Thu, 3 Aug 2017 09:16:54 +0300
Subject: [PATCH 078/256] cosmetic: bootcount: correct comment for used scratch
 register

Signed-off-by: Tomas Melin <tomas.melin@vaisala.com>
---
 drivers/bootcount/bootcount_davinci.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/bootcount/bootcount_davinci.c b/drivers/bootcount/bootcount_davinci.c
index fa87b5e7b9..17829beaa2 100644
--- a/drivers/bootcount/bootcount_davinci.c
+++ b/drivers/bootcount/bootcount_davinci.c
@@ -18,9 +18,9 @@ void bootcount_store(ulong a)
 		(struct davinci_rtc *)CONFIG_SYS_BOOTCOUNT_ADDR;
 
 	/*
-	 * write RTC kick register to enable write
-	 * for RTC Scratch registers. Scratch0 and 1 are
-	 * used for bootcount values.
+	 * write RTC kick registers to enable write
+	 * for RTC Scratch registers. Scratch register 2 is
+	 * used for bootcount value.
 	 */
 	writel(RTC_KICK0R_WE, &reg->kick0r);
 	writel(RTC_KICK1R_WE, &reg->kick1r);
-- 
2.13.6

