From 4f1295b960bb783d6911f535ad9159cea6bb36a4 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 3 Feb 2022 04:32:12 +0800
Subject: [PATCH 36/36] default 1A for rk818 charger

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/power/supply/rk818_charger.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/supply/rk818_charger.c b/drivers/power/supply/rk818_charger.c
index d4ed7b1d558a..c85b3f961636 100644
--- a/drivers/power/supply/rk818_charger.c
+++ b/drivers/power/supply/rk818_charger.c
@@ -234,7 +234,7 @@ static void rk818_usb_power_external_power_changed(struct power_supply *psy)
 {
         struct rk818_charger *cg = power_supply_get_drvdata(psy);
 
-	power_supply_set_input_current_limit_from_supplier_min(cg->usb_psy, 500000);
+	power_supply_set_input_current_limit_from_supplier_min(cg->usb_psy, 1000000);
 }
 
 static enum power_supply_property rk818_usb_power_props[] = {
-- 
2.30.2

