diff -Naur linux-4.11.6.old/arch/arm/boot/dts/sun8i-h3-bananapi-m2-plus.dts linux-4.11.6/arch/arm/boot/dts/sun8i-h3-bananapi-m2-plus.dts
--- linux-4.11.6.old/arch/arm/boot/dts/sun8i-h3-bananapi-m2-plus.dts	2017-07-07 10:29:15.962506659 +0800
+++ linux-4.11.6/arch/arm/boot/dts/sun8i-h3-bananapi-m2-plus.dts	2017-07-07 10:48:58.335852506 +0800
@@ -53,6 +53,7 @@
 	compatible = "sinovoip,bpi-m2-plus", "allwinner,sun8i-h3";
 
 	aliases {
+		ethernet0 = &emac;
 		serial0 = &uart0;
 		serial1 = &uart1;
 	};
