--- linux-4.11-rc7/arch/arm/boot/dts/sun8i-a23-a33.dtsi	2017-04-17 08:40:52.817459890 +0800
+++ linux-4.11-rc7.new/arch/arm/boot/dts/sun8i-a23-a33.dtsi	2017-04-17 08:40:25.514052974 +0800
@@ -119,6 +119,20 @@
 		};
 	};
 
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+		
+		cma: linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x4000000>;
+			alignment = <0x2000>;
+			linux,cma-default;
+		};
+	};
+
 	soc@01c00000 {
 		compatible = "simple-bus";
 		#address-cells = <1>;
@@ -493,6 +507,7 @@
 			clocks = <&ccu CLK_BUS_GPU>, <&ccu CLK_GPU>;
 			clock-names = "bus", "core";
 			resets = <&ccu RST_BUS_GPU>;
+			memory-region = <&cma>;
 
 			assigned-clocks = <&ccu CLK_GPU>;
 			assigned-clock-rates = <384000000>;
