--- linux-4.11.old/arch/arm/boot/dts/sun8i-h3-bananapi-m2-plus.dts	2017-05-01 10:47:48.000000000 +0800
+++ linux-4.11/arch/arm/boot/dts/sun8i-h3-bananapi-m2-plus.dts	2017-05-05 21:16:40.482503049 +0800
@@ -93,6 +93,10 @@
 	};
 };
 
+&de {
+	status = "okay";
+};
+
 &ehci1 {
 	status = "okay";
 };
@@ -101,6 +105,10 @@
 	status = "okay";
 };
 
+&hdmi {
+	status = "okay";
+};
+
 &ir {
 	pinctrl-names = "default";
 	pinctrl-0 = <&ir_pins_a>;
@@ -171,6 +179,10 @@
 	};
 };
 
+&tcon0 {
+	status = "okay";
+};
+
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_pins_a>;
